aws/resource_aws_s3_bucket.go

```diff
@@ -2777,13 +2784,76 @@ func flattenS3ObjectLockConfiguration(conf *s3.ObjectLockConfiguration) []interf
+	for _, granteeObject := range ap.Grants {
+		if aws.StringValue(granteeObject.Grantee.ID) == aws.StringValue(ap.Owner.ID) &&
+			aws.StringValue(granteeObject.Permission) == s3.PermissionFullControl {
+			owner_full_control = true
+			continue
+		}
+		if aws.StringValue(granteeObject.Grantee.URI) == "http://acs.amazonaws.com/groups/global/AllUsers" &&
+			aws.StringValue(granteeObject.Permission) == s3.PermissionRead {
+			all_users_read = true
+			continue
+		}
+		if aws.StringValue(granteeObject.Grantee.URI) == "http://acs.amazonaws.com/groups/global/AllUsers" &&
+			aws.StringValue(granteeObject.Permission) == s3.PermissionWrite {
+			all_users_write = true
+			continue
+		}
+		if aws.StringValue(granteeObject.Grantee.URI) == "http://acs.amazonaws.com/groups/global/AuthenticatedUsers" &&
+			aws.StringValue(granteeObject.Permission) == s3.PermissionRead {
+			authenticated_users_read = true
+			continue
+		}
+		// This is the Amazon ID for the za-team group used for EC2 operations in S3 buckets
+		if aws.StringValue(granteeObject.Grantee.ID) == "6aa5a366c34c1cbe25dc49211496e913e0351eb0e8c37aa3477e40942ec6b97c" &&
+			aws.StringValue(granteeObject.Permission) == s3.PermissionRead {
+			ec2_read = true
+			continue
+		}
+		if aws.StringValue(granteeObject.Grantee.URI) == "http://acs.amazonaws.com/groups/s3/LogDelivery" &&
+			aws.StringValue(granteeObject.Permission) == s3.PermissionReadAcp {
+			logdelivery_read_acp = true
+			continue
+		}
+		if aws.StringValue(granteeObject.Grantee.URI) == "http://acs.amazonaws.com/groups/s3/LogDelivery" &&
+			aws.StringValue(granteeObject.Permission) == s3.PermissionWrite {
+			logdelivery_write = true
+			continue
+		}
+
+		// This is a non standard grant combination, so it's not a canned ACL
+		return false
 	}
 
+	switch {
+	case owner_full_control && !all_users_read && !all_users_write && !authenticated_users_read && !ec2_read && !logdelivery_read_acp && !logdelivery_write:
+		d.Set("acl", "private")
+	case owner_full_control && all_users_read && !all_users_write && !authenticated_users_read && !ec2_read && !logdelivery_read_acp && !logdelivery_write:
+		d.Set("acl", "public-read")
+	case owner_full_control && all_users_read && all_users_write && !authenticated_users_read && !ec2_read && !logdelivery_read_acp && !logdelivery_write:
+		d.Set("acl", "public-read-write")
+	case owner_full_control && !all_users_read && !all_users_write && authenticated_users_read && !ec2_read && !logdelivery_read_acp && !logdelivery_write:
+		d.Set("acl", "authenticated-read")
+	case owner_full_control && !all_users_read && !all_users_write && !authenticated_users_read && ec2_read && !logdelivery_read_acp && !logdelivery_write:
+		d.Set("acl", "aws-exec-read")
+	case owner_full_control && !all_users_read && !all_users_write && !authenticated_users_read && !ec2_read && logdelivery_read_acp && logdelivery_write:
+		d.Set("acl", "log-delivery-write")
+	}
```

Comment:
The reference for these is at https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html#canned-acl